frontend/src/component/admin/auth/SsoGroupSettings.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { Fragment } from 'react';
+import { FormControlLabel, Grid, Switch, TextField } from '@mui/material';
+
+interface SsoGroupSettingsProps {
+    ssoType: string;
+    data?: {
+        enabled: boolean;
+        enableGroupSyncing: boolean;
+        groupJsonPath: string;
+    };
+    setValue: (name: string, value: string | boolean) => void;
+}
+
+export const SsoGroupSettings = ({
+    ssoType,
+    data = {
+        enabled: false,
+        enableGroupSyncing: false,
+        groupJsonPath: '',
+    },
+    setValue,
+}: SsoGroupSettingsProps) => {
+    const updateGroupSyncing = () => {
+        setValue('enableGroupSyncing', !data.enableGroupSyncing);
+    };
+
+    const updateField = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setValue(event.target.name, event.target.value);
+    };
+
+    return (
+        <Fragment>
+            <Grid container spacing={3} mb={2}>
+                <Grid item md={5}>
+                    <strong>Enable Group Syncing</strong>
+                    <p>
+                        Enables automatically syncing of users from the{' '}
+                        {ssoType}
+                        provider when a user logs in.
+                    </p>
+                </Grid>
+                <Grid item md={6} style={{ padding: '20px' }}>
+                    <FormControlLabel
+                        control={
+                            <Switch
+                                onChange={updateGroupSyncing}
+                                value={data.enableGroupSyncing}
+                                name="enableGroupSyncing"
+                                checked={data.enableGroupSyncing}
+                                disabled={!data.enabled}
+                            />
+                        }
+                        label={data.enableGroupSyncing ? 'Enabled' : 'Disabled'}
+                    />
+                </Grid>
+            </Grid>
+            <Grid container spacing={3} mb={2}>
+                <Grid item md={5}>
+                    <strong>Group Field JSON Path</strong>
+                    <p>
+                        Specifies the path in the {ssoType} token response from
```

Comment:
This kind of templated user-facing string literal is going to be a huge pain in the butt when we eventually do i18n but maybe it's okay for now? Or do we have a better solution in the code that I missed?

Reply:
From my experience templated i18n strings are a common use-case, so I'm sure this will be fine when we get to do i18n: https://kazupon.github.io/vue-i18n/guide/formatting.html

Reply:
Sets my mind at ease! If you're confident I'm confident!